pkg/monitor/dynkeepalived.go

```diff
@@ -63,6 +63,32 @@ func KeepalivedWatch(kubeconfigPath, clusterConfigPath, templatePath, cfgPath st
 				}
 			}
 			prevConfig = &newConfig
+
+			// Signal to keepalived whether the haproxy firewall rule is in place
+			ruleExists, err := checkHAProxyPreRoutingRule(apiVip.String(), apiPort, lbPort)
+			if err != nil {
+				log.Error("Failed to check for haproxy firewall rule")
+			} else {
+				filePath := "/var/run/keepalived/iptables-rule-exists"
+				if ruleExists {
+					// FIXME: DRY this
+					_, err := os.Stat(filePath)
+					if os.IsNotExist(err) {
+						_, err := os.Create(filePath)
+						if err != nil {
+							log.WithFields(logrus.Fields{"path": filePath}).Error("Failed to create file")
+						}
+					}
+				} else {
+					_, err := os.Stat(filePath)
```

Comment:
Can't we just call to os.remove and consider err=nil and err = file not exist as OK?

Reply:
Probably, but once I DRY the file existence logic it's one more if statement versus a second clause in another. Plus this way the logic for both creation and removal is the same.